src/test/java/fi/helsinki/cs/tmc/core/commands/DownloadAdaptiveExerciseTest.java

```diff
@@ -121,13 +133,14 @@ private void setUpMocks() throws Exception {
     @Test
     public void testDownloadAndExtractSuccessWithRealZip() throws Exception {
         verifyZeroInteractions(langs);
-        TmcServerCommunicationTaskFactory realFactory = new TmcServerCommunicationTaskFactory();
+        TmcServerCommunicationTaskFactory realFactory = new TmcServerCommunicationTaskFactory(settings, oauth);
         assertNotNull(TmcSettingsHolder.get());
         command = new DownloadAdaptiveExercise(mockObserver, realFactory);
+        Path testPath = Paths.get(System.getProperty("user.dir"));
```

Comment:
You have this at the start of the file:
```
@Rule
public TemporaryFolder testFolder = new TemporaryFolder();
```

Use that (or other temporary folders, created f.ex. by `Files.createTempDirectory(...)`) rather than doing things inside the user's home. `user.dir` is not a proper location for temporary files.